app/gui/src/controller/searcher.rs

```diff
@@ -1235,16 +1244,63 @@ impl EditMetadataGuard {
             }),
         )
     }
+
+    fn get_saved_expression(&self) -> FallibleResult<Option<NodeEditStatus>> {
+        let module = &self.graph.graph().module;
+        let mut edit_status = None;
+        module.with_node_metadata(
+            self.node_id,
+            Box::new(|m| {
+                edit_status = m.edit_status.clone();
+            }),
+        )?;
+        Ok(edit_status)
+    }
+
+    fn revert_node_expression(&self) -> FallibleResult {
+        let edit_status = self.get_saved_expression()?;
+        match edit_status {
+            None => {
+                tracing::warn!("Tried to revert edited node, but found not edit metadata.");
+            }
+            Some(NodeEditStatus::Created) => {
+                tracing::debug!("Deleting node {} after edit reverted.", self.node_id);
+                self.graph.graph().remove_node(self.node_id)?;
+            }
+            Some(NodeEditStatus::Edited { previous_expression }) => {
+                tracing::debug!(
+                    "Reverting node {} after edit reverted. Expression to revert: {}",
```

Comment:
What does mean "Reverting node {} after edit reverted"? I do not understand this sentence TBH. Also, is the "Expression to revert" the old one or the new one?